### PR TITLE
Fix config unit test on macOS and Windows

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Adapted unit tests when running in macOS and Windows

## Why?

Fixing CI

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

**Comment by mlcdf** [comment]
```
config_test.go:46: expected *os.PathError, got *fs.PathError
```
On unix, we have a `*os.PathError` and on windows a `*fs.PathError`. wtf

**Comment by feedmeapples** [comment]
> ```
> config_test.go:46: expected *os.PathError, got *fs.PathError
> ```
> 
> On unix, we have a `*os.PathError` and on windows a `*fs.PathError`. wtf

updated

**Comment by mlcdf** [comment]
```
config_test.go:48: expected *fs.PathError, got *fs.PathError
```
This keeps getting weirder haha

Via https://pkg.go.dev/errors#Is maybe it works?

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -39,11 +41,20 @@ const (
...
 	_, err := config.NewConfig(appName, "../../../../../../../../../../../../tmp")
 
-	if assert.Error(t, err) {
+	if runtime.GOOS == "windows" {
+		if e, ok := err.(*fs.PathError); !ok {
+			t.Errorf("expected *fs.PathError, got %T", e)
+		}
+	} else {
+		if e, ok := err.(*os.PathError); !ok {
+			t.Errorf("expected *os.PathError, got %T", e)
```
Comment (mlcdf) [inline pkg/config/config_test.go:52]: shouldn't we print the type of `err` instead of `e` here?

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -39,11 +41,20 @@ const (
...
+		}
+	}
+
+	if runtime.GOOS == "linux" {
+		expectedError := errors.New("open /tmp.yaml: permission denied")
 		assert.Equal(t, expectedError.Error(), err.Error())
```
Comment (bergundy) [inline pkg/config/config_test.go:58]: nit: use `errors.Is`

File: pkg/config/config_test.go [hunk 3]
```diff
@@ -39,11 +41,20 @@ const (
...
 
 func TestNewConfigPermissionDenied(t *testing.T) {
-	expectedError := errors.New("open /tmp.yaml: permission denied")
-
 	_, err := config.NewConfig(appName, "../../../../../../../../../../../../tmp")
 
-	if assert.Error(t, err) {
+	if runtime.GOOS == "windows" {
+		if e, ok := err.(*fs.PathError); !ok {
+			t.Errorf("expected *fs.PathError, got %T", e)
```
Comment (bergundy) [inline pkg/config/config_test.go:48]: Would be nice to have a stricter test here

File: pkg/config/config_test.go [hunk 4]
```diff
@@ -39,12 +41,26 @@ const (
 )
 
 func TestNewConfigPermissionDenied(t *testing.T) {
-	expectedError := errors.New("open /tmp.yaml: permission denied")
-
 	_, err := config.NewConfig(appName, "../../../../../../../../../../../../tmp")
```
Comment (feedmeapples) [inline pkg/config/config_test.go:44]: the test worked on a local windows env. testing with more subpaths here 

```suggestion
	_, err := config.NewConfig(appName, "../../../../../../../../../../../../../../../../../tmp")
```

**Comment by feedmeapples** [comment]
i've ingored permission denied on windows check as it was resulting in a config file `C:\tmp.yaml` and just no error

**Review by mlcdf — APPROVED** [review]
Good idea!

File: pkg/config/config_test.go [hunk 5]
```diff
@@ -38,15 +41,33 @@ const (
...
-
-	cfg, err := config.NewConfig(appName, "../../../../../../../../../../../../tmp")
+	dir, err := os.UserHomeDir()
 	assert.NoError(t, err)
 
-	err = cfg.SetEnvProperty("test", "test", "test")
+	appName := uuid.New()
+	readOnly := os.FileMode(0400)
+	dir = filepath.Join(dir, ".config", appName)
+	os.MkdirAll(dir, readOnly) // owner read only
```
Comment (feedmeapples) [inline pkg/config/config_test.go:50]:
```suggestion
	os.MkdirAll(dir, readOnly)
```

File: pkg/config/config_test.go [hunk 6]
```diff
@@ -38,15 +41,33 @@ const (
...
+	dir, err := os.UserHomeDir()
 	assert.NoError(t, err)
 
-	err = cfg.SetEnvProperty("test", "test", "test")
+	appName := uuid.New()
+	readOnly := os.FileMode(0400)
+	dir = filepath.Join(dir, ".config", appName)
+	os.MkdirAll(dir, readOnly)
+
+	// umask may have changed the file permissions, ensure they are correct
```
Comment (feedmeapples) [inline pkg/config/config_test.go:52]:
```suggestion
	// umask may have changed config folder permissions, ensure they are correct
```

File: pkg/config/config_test.go [hunk 7]
```diff
@@ -38,15 +41,33 @@ const (
...
+	err = os.Chmod(dir, os.FileMode(readOnly))
+	assert.NoError(t, err)
 
-	if assert.Error(t, err) {
-		assert.Equal(t, expectedError.Error(), err.Error())
+	_, err = config.NewConfig(appName, "test")
+
+	switch runtime.GOOS {
+	case "windows":
+		t.Skip("Emulating permission denied on Windows results in C:\tmp.yaml and no error")
```
Comment (feedmeapples) [inline pkg/config/config_test.go:60]:
```suggestion
		t.Skip("no permission denied error on Windows")
```

**Review by robholland — APPROVED** [review]

**Pull request merged**